docs/inference-providers/index.md

```diff
@@ -144,71 +149,381 @@ completion = client.chat.completions.create(
+</hfoption>
+
+</hfoptions>
+
+#### HTTP / cURL
+
+For testing, debugging, or integrating with any HTTP client, here's the raw REST API format. Our intelligent routing automatically selects the optimal provider for your requested model:
+
+```bash
+curl https://router.huggingface.co/v1/chat/completions \
+    -H "Authorization: Bearer $HF_TOKEN" \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "messages": [
+            {
+                "role": "user",
+                "content": "How many G in huggingface?"
+            }
+        ],
+        "model": "deepseek-ai/DeepSeek-V3-0324",
+        "stream": false
+    }'
+```
+
+### Quick Start - Text-to-Image Generation
+
+Let's explore how to generate images from text prompts using Inference Providers. We'll use [black-forest-labs/FLUX.1-dev](https://huggingface.co/black-forest-labs/FLUX.1-dev), a state-of-the-art diffusion model that produces highly detailed, photorealistic images.
+
+#### Python
+
+Use the `huggingface_hub` library for the simplest image generation experience with automatic provider selection:
+
+```python
+import os
+from huggingface_hub import InferenceClient
+
+client = InferenceClient(api_key=os.environ["HF_TOKEN"])
+
+image = client.text_to_image(
+    prompt="A serene lake surrounded by mountains at sunset, photorealistic style",
+    model="black-forest-labs/FLUX.1-dev"
+)
+
+# Save the generated image
+image.save("generated_image.png")
+```
+
+#### JavaScript
+
+Use our JavaScript SDK for streamlined image generation with TypeScript support:
 
 ```js
 import { InferenceClient } from "@huggingface/inference";
+import fs from "fs";
 
 const client = new InferenceClient(process.env.HF_TOKEN);
 
-const chatCompletion = await client.chatCompletion({
-    provider: "novita",
-    model: "deepseek-ai/DeepSeek-V3-0324",
-    messages: [
-        {
-            role: "user",
-            content: "How many 'G's in 'huggingface'?",
-        },
-    ],
+const imageBlob = await client.textToImage({
+  model: "black-forest-labs/FLUX.1-dev",
+  inputs:
+    "A serene lake surrounded by mountains at sunset, photorealistic style",
 });
 
-console.log(chatCompletion.choices[0].message);
+// Save the image
+const buffer = Buffer.from(await imageBlob.arrayBuffer());
+fs.writeFileSync("generated_image.png", buffer);
+```
+
+## Provider Selection
+
+The Inference Providers API acts as a unified proxy layer that sits between your application and multiple AI providers. Understanding how provider selection works is crucial for optimizing performance, cost, and reliability in your applications.
+
+### API as a Proxy Service
+
+When using Inference Providers, your requests go through Hugging Face's proxy infrastructure, which provides several key benefits:
+
+- **Unified Authentication & Billing**: Use a single Hugging Face token for all providers
+- **Automatic Failover**: If one provider is unavailable, requests can be routed to alternatives
```

Comment:
not true

Reply:
ah yes sorry, I understood it as "if request fails, it's rerouted to the next one"